### PR TITLE
sql: add switch for check constraint

### PR DESCRIPTION
**Pull request by fzzf678**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v7.2 (TiDB 7.2 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v7.0 (TiDB 7.0 versions)
- [ ] v6.6 (TiDB 6.6 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:https://github.com/pingcap/docs-cn/pull/14320
- Other reference link(s): https://github.com/pingcap/tidb/issues/41711

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: constraints.md [hunk 1]
```diff
@@ -52,6 +52,11 @@ Query OK, 1 row affected (0.03 sec)
 
 ## CHECK
 
+> **Note:**
+>
+> 'CHECK' constraint function is off by default, [`tidb_enable_check_constraint`](/system-variables.md#tidb_enable_check_constraint-New in v7.2.0) needs to be enabled before it takes effect.
```
Comment (qiancai) [inline constraints.md:57]:
```suggestion
> The 'CHECK' constraint feature is disabled by default. To enable it, You need to set the [`tidb_enable_check_constraint`](/system-variables.md#tidb_enable_check_constraint-new-in-v720) variable to `ON`.
```

File: constraints.md [hunk 2]
```diff
@@ -129,6 +134,11 @@ In addition to specifying `[NOT] ENFORCED` when adding the constraint, you can a
 ALTER TABLE t ALTER CONSTRAINT c1 NOT ENFORCED;
 ```
 
+### MySQL Compatibility
```
Comment (qiancai) [inline constraints.md:137]:
```suggestion
### MySQL compatibility
```

File: constraints.md [hunk 3]
```diff
@@ -129,6 +134,11 @@ In addition to specifying `[NOT] ENFORCED` when adding the constraint, you can a
...
+
+- It is not supported to add a `CHECK` constraint while adding a column in (for example, `ALTER TABLE t ADD COLUMN a CHECK(a > 0)`), otherwise only the column will be added successfully, and TiDB will ignore the `CHECK` constraint and will not report an error.
```
Comment (qiancai) [inline constraints.md:139]:
```suggestion
- It is not supported to add a `CHECK` constraint while adding a column (for example, `ALTER TABLE t ADD COLUMN a CHECK(a > 0)`). In this case, only the column will be added successfully, and TiDB will ignore the `CHECK` constraint without reporting any error.
```

File: constraints.md [hunk 4]
```diff
@@ -129,6 +134,11 @@ In addition to specifying `[NOT] ENFORCED` when adding the constraint, you can a
...
+- It is not supported to use `ALTER TABLE t CHANGE a b int CHECK(b > 0)` to add a `CHECK` constraint, and TiDB will report an error when using this statement.
```
Comment (qiancai) [inline constraints.md:140]:
```suggestion
- It is not supported to use `ALTER TABLE t CHANGE a b int CHECK(b > 0)` to add a `CHECK` constraint. When this statement is executed, TiDB will report an error.
```

File: system-variables.md [hunk 5]
```diff
@@ -1616,6 +1616,13 @@ mysql> SELECT job_info FROM mysql.analyze_jobs ORDER BY end_time DESC LIMIT 1;
 - Default value: `OFF`
 - This variable is used to control whether to enable the cascades planner.
 
+### tidb_enable_check_constraint <span class="version-mark">New in v7.2.0</span>
+
+- Scope: GLOBAL
```
Comment (qiancai) [inline system-variables.md:1621]:
```suggestion
- Scope: GLOBAL
- Persists to cluster: Yes
```

File: system-variables.md [hunk 6]
```diff
@@ -1616,6 +1616,13 @@ mysql> SELECT job_info FROM mysql.analyze_jobs ORDER BY end_time DESC LIMIT 1;
...
+- Type: Boolean
+- Default value: `OFF`
+- This variable is used to control whether to enable the [`CHECK` constraint](/constraints.md#check).
```
Comment (qiancai) [inline system-variables.md:1624]:
```suggestion
- This variable is used to control whether to enable the [`CHECK` constraint](/constraints.md#check) feature.
```

**Review by qiancai — APPROVED** [review]
/lgtm

File: constraints.md [hunk 7]
```diff
@@ -52,6 +52,11 @@ Query OK, 1 row affected (0.03 sec)
 
 ## CHECK
 
+> **Note:**
+>
+> The 'CHECK' constraint feature is disabled by default. To enable it, You need to set the [`tidb_enable_check_constraint`](/system-variables.md#tidb_enable_check_constraint-new-in-v720) variable to `ON`.
```
Comment (qiancai) [inline constraints.md:57]:
```suggestion
> The 'CHECK' constraint feature is disabled by default. To enable it, you need to set the [`tidb_enable_check_constraint`](/system-variables.md#tidb_enable_check_constraint-new-in-v720) variable to `ON`.
```

File: constraints.md [hunk 8]
```diff
@@ -52,6 +52,11 @@ Query OK, 1 row affected (0.03 sec)
 
 ## CHECK
 
+> **Note:**
+>
+> The 'CHECK' constraint feature is disabled by default. To enable it, you need to set the [`tidb_enable_check_constraint`](/system-variables.md#tidb_enable_check_constraint-new-in-v720) variable to `ON`.
+
```
Comment (qiancai) [inline constraints.md:58]:
```suggestion
```

File: constraints.md [hunk 9]
```diff
@@ -52,6 +52,10 @@ Query OK, 1 row affected (0.03 sec)
 
 ## CHECK
 
+> **Note:**
+>
+> The 'CHECK' constraint feature is disabled by default. To enable it, you need to set the [`tidb_enable_check_constraint`](/system-variables.md#tidb_enable_check_constraint-new-in-v720) variable to `ON`.
```
Comment (Oreoxmt) [inline constraints.md:57]:
```suggestion
> The `CHECK` constraint feature is disabled by default. To enable it, you need to set the [`tidb_enable_check_constraint`](/system-variables.md#tidb_enable_check_constraint-new-in-v720) variable to `ON`.
```

File: constraints.md [hunk 10]
```diff
@@ -129,6 +133,11 @@ In addition to specifying `[NOT] ENFORCED` when adding the constraint, you can a
 ALTER TABLE t ALTER CONSTRAINT c1 NOT ENFORCED;
 ```
 
+### MySQL compatibility
+
+- It is not supported to add a `CHECK` constraint while adding a column (for example, `ALTER TABLE t ADD COLUMN a CHECK(a > 0)`). In this case, only the column will be added successfully, and TiDB will ignore the `CHECK` constraint without reporting any error.
+- It is not supported to use `ALTER TABLE t CHANGE a b int CHECK(b > 0)` to add a `CHECK` constraint. When this statement is executed, TiDB will report an error.
```
Comment (Oreoxmt) [inline constraints.md:139]:
```suggestion
- It is not supported to add a `CHECK` constraint while adding a column (for example, `ALTER TABLE t ADD COLUMN a CHECK(a > 0)`). In this case, only the column is added successfully, and TiDB ignores the `CHECK` constraint without reporting any error.
- It is not supported to use `ALTER TABLE t CHANGE a b int CHECK(b > 0)` to add a `CHECK` constraint. When this statement is executed, TiDB reports an error.
```

**Pull request merged**